File: Prac2-25/4.py
def isLeap(year):
    return (year % 4 == 0 and year % 100 != 0) or year % 400 == 0

for _ in range(int(input())):
    y1 = int(input())
    counter = 0
    for year in range(y1 + 1, 3000):
        if (isLeap(year)):
            counter += 2
        else:
            counter += 1

        if (counter % 7 == 0 and (isLeap(year) == isLeap(y1))):
            print(f'Calendar for {y1} can next be reused in {year}.')
            break

File: Prac2-25/4.cpp
#include <iostream>
using namespace std;

int main()
{
    int n; 
    cin >>n; 
    for(int i=1; i<=n; i++)
    {
        int y;
        cin >> y;
        int num = 0;
        if (y%4 == 0 && y%100 != 0 || y%400 == 0)
            num = 1;
        else
            num = 0;

        if (num == 0)
        {
            if (y%4 ==0 || (y-1)%4 ==0)
                cout << "Calendar for " << y << " can next be reused in " << y+6 << "." << endl;
            else
                cout << "Calendar for " << y << " can next be reused in " << y+11 << "." << endl;
        }
            
        else
            cout << "Calendar for " << y << " can next be reused in " << y+28 << "." << endl;
    }
    return 0;
}